engine/src/cubos/engine/splitscreen/splitscreen.cpp

```diff
@@ -0,0 +1,41 @@
+#include <cubos/engine/renderer/plugin.hpp>
+#include <cubos/engine/renderer/renderer.hpp>
+#include <cubos/engine/splitscreen/splitscreen.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Splits the viewport recursively for the given cameras.
+    /// @param position Viewport position.
+    /// @param size Viewport size.
+    /// @param count How many cameras need to be fitted in to the given viewport.
+    /// @param activeCameras Output array where the viewports will be set.
+    void setViewportCameras(glm::ivec2 position, glm::ivec2 size, int count, glm::ivec2* positions, glm::ivec2* sizes)
+    {
+        if (count == 1)
+        {
+            positions[0] = position;
+            sizes[0] = size;
+        }
+        else if (count >= 2)
+        {
+            glm::ivec2 splitSize;
+            glm::ivec2 splitOffset;
+
+            // Split along the largest axis.
+            if (size.x > size.y)
+            {
+                splitSize = {size.x / 2, size.y};
+                splitOffset = {size.x / 2, 0};
+            }
+            else
+            {
+                splitSize = {size.x, size.y / 2};
+                splitOffset = {0, size.y / 2};
+            }
+
+            setViewportCameras(position, splitSize, count / 2, positions, sizes);
+            setViewportCameras(position + splitOffset, splitSize, (count + 1) / 2, &positions[count / 2],
+                               &sizes[count / 2]);
+        }
+    }
```

Comment:
I would make this a `static` function inside `splitscreen/plugin.cpp`, as it is never used out of there.